babyai/bot.py

```diff
@@ -150,25 +143,19 @@ def foolish_action_while_exploring(self, action):
                 # need to open it
                 self.bot.stack.append(OpenSubgoal(self.bot))
 
-    def take_action(self, action):
-        """
-        Function that updates the bot's stack given the action played
-        Should be overridden in all sub-classes
-        TODO: There are some steps that are common in both take_action and get_action for certain subgoals, maybe the bot's speed can be improved if we do them only once - or at least we can factorize the code a bit
+    def replan(self, given_action):
```

Comment:
Here `given_action` is the action the agent actually took, as opposed to what the bot suggested?

If so, I would call it `action_taken`, because `given_action` sounds like the action the bot gave.